plugin/storage/grpc/factory.go

```diff
@@ -98,7 +107,8 @@ func (f *Factory) Initialize(metricsFactory metrics.Factory, logger *zap.Logger)
 		for _, opt := range opts {
 			clientOpts = append(clientOpts, configgrpc.WithGrpcDialOption(opt))
 		}
-		return f.config.ToClientConn(context.Background(), componenttest.NewNopHost(), telset, clientOpts...)
+		noophost := NewNopHost()
```

Comment:
we also have a usage of `componenttest.NewNopHost` in https://github.com/jaegertracing/jaeger/blob/main/cmd/jaeger/internal/integration/span_writer.go#L55. Maybe we can move the `NopHost` to a shared package rather than just having it live here so we don't have to define it every time we need it?

Reply:
the point was to use the real Host provided by OTEL collector. There's no reason to define our own noop host.

Reply:
looks like we have another noop host in jaeger as well - https://github.com/jaegertracing/jaeger/blob/main/cmd/collector/app/handler/otlp_receiver.go#L185

Reply:
So we need host implementation here with its related methods ?